ui/tract-renderer/src/main/java/org/cru/godtools/tract/util/LottieUtils.kt

```diff
@@ -0,0 +1,20 @@
+package org.cru.godtools.tract.util
+
+import android.content.Context
+import android.util.TypedValue
+import android.view.MenuItem
+import androidx.annotation.RawRes
+import com.airbnb.lottie.LottieCompositionFactory
+import com.airbnb.lottie.LottieDrawable
+
+fun MenuItem.loadAnimation(context: Context, @RawRes rawRes: Int) {
+    LottieCompositionFactory.fromRawRes(context, rawRes).addListener { comp ->
+        icon = LottieDrawable().apply {
+            composition = comp
+            repeatCount = LottieDrawable.INFINITE
+            val size = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 24f, context.resources.displayMetrics)
+            scale = comp.scaleTo(size, size)
+            playAnimation()
+        }
+    }
+}
```

Comment:
Similar thing here, are you creating a new method on ```MenuItem```? Also:

1. Where are the values in the size value coming from(TypedValue.COMPLEX_UNIT_DIP, etc)
2. Composition is equal to the lottie composition itself, which is the same as the animation right? So when you do ```comp.scaleTo```, you're referring to the method you added/created, right?
3. ```playAnimation()``` is just a method from the lottie library right?

Reply:
1. action icons are supposed to be `24dp` x `24dp`, the `TypedValue` method will resolve the raw pixel size utilizing the `resources.displayMetrics`.
2. yeah, `LottieComposition` represents a loaded animation (`LottieCompositionFactory.fromRawRes()`), calling the `scaleTo()` extension method will calculate how to scale the animation to fit within the 24 x 24 display size.
3. yeah, the method is `LottieDrawable.playAnimation()`. I'm using `.apply` on line 12, which makes `this` the `LottieDrawable`, and I can call methods on `this` without needing to directly reference the object